packages/go_router_builder/example/lib/all_types.dart

```diff
@@ -342,6 +344,77 @@ class IterableRoute extends GoRouteData {
       );
 }
 
+class IterableRouteWithDefaultValues extends GoRouteData {
+  const IterableRouteWithDefaultValues({
+    this.intIterableField = const <int>[0],
+    this.doubleIterableField = const <double>[0, 1, 2],
+    this.stringIterableField = const <String>['defaultValue'],
+    this.boolIterableField = const <bool>[false],
+    this.enumIterableField = const <SportDetails>[
+      SportDetails.tennis,
+      SportDetails.hockey
+    ],
+    this.intListField = const <int>[0],
+    this.doubleListField = const <double>[1, 2, 3],
+    this.stringListField = const <String>['defaultValue0', 'defaultValue1'],
+    this.boolListField = const <bool>[true],
+    this.enumListField = const <SportDetails>[SportDetails.football],
+    this.intSetField = const <int>{0, 1},
+    this.doubleSetField = const <double>{},
+    this.stringSetField = const <String>{'defaultValue'},
+    this.boolSetField = const <bool>{true, false},
+    this.enumSetField = const <SportDetails>{SportDetails.hockey},
+  });
+
+  final Iterable<int> intIterableField;
+  final List<int> intListField;
+  final Set<int> intSetField;
+
+  final Iterable<double> doubleIterableField;
+  final List<double> doubleListField;
+  final Set<double> doubleSetField;
+
+  final Iterable<String> stringIterableField;
+  final List<String> stringListField;
+  final Set<String> stringSetField;
+
+  final Iterable<bool> boolIterableField;
+  final List<bool> boolListField;
+  final Set<bool> boolSetField;
+
+  final Iterable<SportDetails> enumIterableField;
+  final List<SportDetails> enumListField;
+  final Set<SportDetails> enumSetField;
+
+  @override
+  Widget build(BuildContext context, GoRouterState state) => BasePage<String>(
```

Comment:
Maybe build a widget that holds all these parameter as its property, and then you can use `tester.widget<>(find.byType)` to get the widget and then you can compare the value of each property in the widget.

Reply:
I've implemented it in [test: Add better tests](https://github.com/flutter/packages/pull/3391/commits/cbfef9d67901501a27c043379d68e209640195c5)